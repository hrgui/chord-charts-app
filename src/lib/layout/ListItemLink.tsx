import * as React from "react";
import { NavLink as ReactRouterNavLink } from "react-router-dom";
import ListItem from "@material-ui/core/ListItem";
import styled from "styled-components";

const activeClassName = `ListItemLink-active`;
const StyledListItem = styled(ListItem).attrs({ activeClassName })`
  &.${activeClassName} {
    background: ${({ theme }) => theme.palette.background.default};
  }
`;

const ListItemLink = ({ ...props }: any) => (
  <StyledListItem button component={ReactRouterNavLink} exact {...props} />
);

export default ListItemLink;
