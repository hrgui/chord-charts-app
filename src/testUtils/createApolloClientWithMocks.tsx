import { ApolloClient } from "@apollo/client";
import { InMemoryCache } from "@apollo/client";

export default function createApolloClientWithMocks(context) {
  return new ApolloClient({
    cache: new InMemoryCache(),
  });
}
