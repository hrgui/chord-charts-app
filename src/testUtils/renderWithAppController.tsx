import React from "react";
import { render, act, wait } from "@testing-library/react";
import { AppController } from "app/core/AppController";

export async function renderWithAppController(
  ui,
  { store, reduxStore, ...appControllerProps }: any = {}
) {
  let el;
  act(() => {
    el = render(
      <AppController
        reduxStore={reduxStore || store}
        testMode
        {...appControllerProps}
      >
        {ui}
      </AppController>
    );
  });
  // AppThemeProvider re-renders twice
  await wait();

  const { rerender: _rerender, ...otherElProps } = el;

  return {
    ...otherElProps,
    rerender: ui => {
      return _rerender(
        <AppController
          reduxStore={reduxStore || store}
          testMode
          {...appControllerProps}
        >
          {ui}
        </AppController>
      );
    }
  };
}
