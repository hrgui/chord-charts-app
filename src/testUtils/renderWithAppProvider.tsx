import React from "react";
import { render, act, RenderResult } from "@testing-library/react";
import { AppProvider } from "app/core/AppProvider";
import { Router as TestRouter } from "react-router-dom";
import { createStore } from "app/store";

export interface RenderWithAppControllerProps {
  store?;
  [name: string]: any;
}

export function renderWithAppController(
  ui,
  {
    store = createStore(),
    gqlMocks = [],
    mockedProviderProps = {},
    ...appControllerProps
  }: RenderWithAppControllerProps = {}
): RenderResult {
  const TestComponentProviderOverrides = {
    Router: appControllerProps.history && TestRouter,
  };

  let el;
  act(() => {
    el = render(
      <AppProvider
        componentProviderOverrides={TestComponentProviderOverrides}
        store={store}
        apolloProviderProps={{ mocks: gqlMocks, ...mockedProviderProps }}
        {...appControllerProps}
      >
        {ui}
      </AppProvider>
    );
  });

  const { rerender: _rerender, ...otherElProps } = el;

  return {
    ...otherElProps,
    rerender: (ui) => {
      return _rerender(
        <AppProvider
          componentProviderOverrides={TestComponentProviderOverrides}
          store={store}
          apolloProviderProps={{ mocks: gqlMocks, ...mockedProviderProps }}
          {...appControllerProps}
        >
          {ui}
        </AppProvider>
      );
    },
  };
}
