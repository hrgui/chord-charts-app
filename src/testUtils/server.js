// src/server.js
import { Server, Model } from "miragejs";
import { buildSchema, graphql } from "graphql";

let graphqlSchema = buildSchema(`
  type Query {
    movies: [Movie]
  }
  type Movie {
    id: ID!
    title: String!
  }
`);

export function makeServer({ environment = "development" } = {}) {
  console.log("i did make the server");
  let server = new Server({
    environment,
    seeds(server) {
      server.create("user", { name: "Bob" });
      server.create("user", { name: "Alice" });
    },

    routes() {
      console.log(" i did call this");

      this.passthrough((req) => {
        console.log(`cypress`, req.url);
        return false;
      });

      this.post("http://localhost:4000/graphql", (schema, request) => {
        let requestJson = JSON.parse(request.requestBody);
        let query = requestJson.query;
        let variables = requestJson.variables;

        let resolver = {
          movies() {
            return schema.db.movies;
          },
        };

        console.log(schema, request);

        return graphql(graphqlSchema, query, resolver, null, variables);
      });
    },
  });

  return server;
}
