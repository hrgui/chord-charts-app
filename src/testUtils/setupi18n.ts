import i18n from "i18next";
import { initReactI18next } from "react-i18next";
const translation = require("../../public/locales/en-US/translation.json");
const setlist = require("../../public/locales/en-US/setlist.json");
const song = require("../../public/locales/en-US/song.json");

i18n.use(initReactI18next).init({
  lng: "en-US",
  fallbackLng: "en-US",
  interpolation: {
    escapeValue: false // not needed for react as it escapes by default
  },
  resources: {
    en: {
      translation,
      setlist,
      song
    }
  }
});
