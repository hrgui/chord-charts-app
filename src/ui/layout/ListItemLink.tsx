import { useAppBarActions } from "~/hooks/useAppBarActions";
import useGetAppBarData from "~/hooks/useGetAppBarData";
import NavLink, { NavLinkProps } from "~/ui/router/NavLink";
import { isMobile } from "utils";

const ListItemLink = ({
  onClick,
  dismissMobileMenu = true,
  ...props
}: NavLinkProps & { dismissMobileMenu?: boolean }) => {
  const { toggleNavMenu } = useAppBarActions();
  const { navMenuHidden } = useGetAppBarData();
  const handleClick = (e) => {
    onClick?.(e);

    //TODO: on mobile navMenuHidden = true is actually shown
    if (dismissMobileMenu && isMobile() && navMenuHidden) {
      toggleNavMenu();
    }
  };

  return (
    <NavLink
      onClick={handleClick}
      activeClassName="bg-base-300"
      className="flex p-2 items-center hover:bg-base-100"
      {...props}
    ></NavLink>
  );
};

export default ListItemLink;
