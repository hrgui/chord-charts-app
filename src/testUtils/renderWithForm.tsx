import { Form } from "lib/form/Form";
import React from "react";
import { renderWithAppController as render } from "./renderWithAppProvider";

interface FormProps {
  initialValues: any;
  onSubmit: any;
  [name: string]: any;
}

export function renderWithForm(
  uiFn,
  formProps: FormProps = { initialValues: {}, onSubmit: () => void 0 }
) {
  const el = render(<Form {...formProps}>{uiFn}</Form>);

  return el;
}
