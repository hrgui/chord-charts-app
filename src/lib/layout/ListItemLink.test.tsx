import React from "react";
import ListItemLink from "./ListItemLink";
import { renderWithAppController as render } from "testUtils/renderWithAppProvider";

test("ListItemLink - default state", async () => {
  const { container } = await render(<ListItemLink to="/test" />);
  expect(container.firstChild).toMatchSnapshot();
});
