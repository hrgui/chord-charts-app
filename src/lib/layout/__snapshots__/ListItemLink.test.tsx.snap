// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListItemLink - default state 1`] = `
<a
  aria-disabled="false"
  class="MuiButtonBase-root MuiListItem-root sc-AxjAm jcYXsi MuiListItem-gutters MuiListItem-button"
  href="/test"
  role="button"
  tabindex="0"
>
  <span
    class="MuiTouchRipple-root"
  />
</a>
`;
